tests/Unit/Domain/Test_OrientationMap.cpp

```diff
@@ -259,3 +260,227 @@ SPECTRE_TEST_CASE("Unit.Domain.OrientationMap", "[Domain][Unit]") {
   test_2d();
   test_3d();
 }
+
+namespace {
+template <size_t VolumeDim>
+OrientationMap<VolumeDim> generate_orientationmap(
+    size_t num_of_cycle_permutation, size_t num_of_sign_permutation) {
+  std::array<size_t, VolumeDim> dimensions{};
+  std::bitset<VolumeDim> bitset = num_of_sign_permutation;
+  std::array<Direction<VolumeDim>, VolumeDim> array_of_directions{};
+  for (size_t i = 0; i < VolumeDim; i++) {
+    gsl::at(dimensions, i) = i;
+  }
+  for (size_t i = 0; i < num_of_cycle_permutation; i++) {
+    std::next_permutation(dimensions.begin(), dimensions.end());
+  }
+  for (size_t i = 0; i < VolumeDim; i++) {
+    gsl::at(array_of_directions, i) = Direction<VolumeDim>{
+        gsl::at(dimensions, i), bitset[i] == 1 ? Side::Upper : Side::Lower};
+  }
+  return OrientationMap<VolumeDim>{array_of_directions};
+}
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.DiscreteRotation.AllOrientations",
+                  "[Domain][Unit]") {
+  for (size_t i = 0; i < 2; i++) {
+    for (size_t j = 0; j < 4; j++) {
+      OrientationMap<2> map_2d = generate_orientationmap<2>(i, j);
+      const std::array<double, 2> original_point{{0.5, -2.0}};
+      const std::array<double, 2> new_point =
+          discrete_rotation(map_2d, original_point);
+      if (map_2d(0) == 0) {
```

Comment:
simplify into only checks with no ifs, and see if you can combine the double, DataVector and reference wrapper tests